src/bundle/settings.rs

```diff
@@ -157,8 +157,17 @@ impl Settings {
             bail!("No [package.metadata.bundle] section in Cargo.toml");
         };
         let (bundle_settings, mut binary_name) = match build_artifact {
-            // TODO: this is wrong. Package can have multiple binaries and none of them has to be named the same way as package itself
-            BuildArtifact::Main => (bundle_settings, package.name.clone()),
+            BuildArtifact::Main => {
+                if let Some(target) = package
+                    .targets
+                    .iter()
+                    .find(|target| target.kind.iter().any(|k| k == "bin"))
```

Comment:
Can this just be `target.kind.contains("bin")`?

Reply:
It causes a compile error because `target.kind` is `Vec<String>`. `target.kind.contains(&"bin".to_string())` works, but it causes redundant heap allocation.